src/Symfony/Component/Console/Helper/TreeHelper.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Helper;
+
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * The TreeHelper class provides methods to display tree-like structures.
+ *
+ * @author Simon André <smn.andre@gmail.com>
+ *
+ * @implements \RecursiveIterator<int, TreeNode>
+ */
+final class TreeHelper implements \RecursiveIterator
+{
+    private readonly TreeStyle $style;
+
+    private readonly TreeNode $node;
+
+    private \Iterator $childrenIterator;
+
+    public function __construct(
```

Comment:
Should this be private if a `create()` method also exists? Having two entrypoints could be confusing, what do you think?

Reply:
Absolutely, thank you!

Refactored into private constructor with readonly ppp & renamed factory `createTree()`  👌